File: feel-engine-wrapper/src/main/java/io/holunda/feel/rest/CreateOutputVariablesResource.java
package io.holunda.feel.rest;

import io.holunda.feel.core.ConnectorHelper;
import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/createOutputVariables")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class CreateOutputVariablesResource {

    @POST
    public Object createOutputVariables(CreateOutputVariablesRequest request) {
        return ConnectorHelper.createOutputVariables(request.context(), request.resultExpression());
    }
}

File: feel-engine-wrapper/src/main/kotlin/io/holunda/feel/rest/CreateOutputVariablesResource.kt
package io.holunda.feel.rest

import io.holunda.feel.core.*
import jakarta.ws.rs.*
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.MediaType.*

@Path("/createOutputVariables")
@Consumes(APPLICATION_JSON)
@Produces(APPLICATION_JSON)
class CreateOutputVariablesResource {

    @POST
    fun createOutputVariables(request: CreateOutputVariablesRequest) =
        FeelEngineWrapper.createOutputVariables(request.context, request.resultExpression)
}

data class CreateOutputVariablesRequest(val context: Any, val resultExpression: String)
